Detections/MultipleDataSources/TimeSeriesAnomaly-MultiVendor_DataExfiltration.yaml

```diff
@@ -53,34 +58,37 @@ query: |
+  | project DeviceVendor, TimeGeneratedHour, TimeGeneratedMax, SourceIP, DestinationIPList, DestinationPortList, TotalSentBytesinMB, TotalReceivedBytesinMB
+  | where TotalSentBytesinMB > bytessentperhourthreshold
   ),
   (
   VMConnection 
   | where isnotempty(DestinationIp) and isnotempty(SourceIp)
   | where TimeGenerated between (startofday(ago(starttime))..startofday(ago(endtime))) 
   | extend DeviceVendor = "VMConnection", SourceIP = SourceIp, DestinationIP = DestinationIp
   | extend SentBytesinMB = ((BytesSent / 1024)/1024), ReceivedBytesinMB = ((BytesReceived / 1024)/1024)
-  | summarize HourlyCount = count(), ListOfDestPorts = make_set(DestinationPort), TotalSentBytesinMB = sum(SentBytesinMB),TotalReceivedBytesinMB = sum(ReceivedBytesinMB) by SourceIP, DestinationIP, DeviceVendor, bin(TimeGenerated,1h)
-  | project DeviceVendor, TimeGenerated, SourceIP, DestinationIP, ListOfDestPorts, TotalSentBytesinMB, TotalReceivedBytesinMB 
+  | summarize HourlyCount = count(),TimeGeneratedMax=arg_max(TimeGenerated, *), DestinationIPList=make_set(DestinationIP), DestinationPortList = make_set(DestinationPort), TotalSentBytesinMB = sum(SentBytesinMB),TotalReceivedBytesinMB = sum(ReceivedBytesinMB) by SourceIP, DestinationIP, DeviceVendor, TimeGeneratedHour=bin(TimeGenerated,1h)
+  | project DeviceVendor, TimeGeneratedHour, TimeGeneratedMax, SourceIP, DestinationIPList, DestinationPortList, TotalSentBytesinMB, TotalReceivedBytesinMB 
+  | where TotalSentBytesinMB > bytessentperhourthreshold
   )
   );
   // Join against base logs to retrive records associated with the hour of anomoly
   TimeSeriesAlerts
   | join (
-      BaseLogs 
-  ) on DeviceVendor, TimeGenerated | sort by score desc 
-  | project DeviceVendor, AnomalyTimeattheHour, SourceIP, DestinationIP, ListOfDestPorts, TotalSentBytesinMB, TotalReceivedBytesinMB, TotalBytesSentinMBperHour, baselinebytessentperHour, score, anomalies 
-  | order by score desc, TotalSentBytesinMB desc
-  | extend IPCustomEntity = DestinationIP
\ No newline at end of file
+      BaseLogs | extend AnomalyHour = TimeGeneratedHour
+  ) on DeviceVendor, AnomalyHour | sort by score desc 
+  | project DeviceVendor, AnomalyHour,TimeGeneratedMax, SourceIP, DestinationIPList, DestinationPortList, TotalSentBytesinMB, TotalReceivedBytesinMB, TotalBytesSentinMBperHour, baselinebytessentperHour, score, anomalies 
+    | summarize count(),TimeGeneratedMax=arg_max(TimeGeneratedMax, *), SourceIPMax= arg_max(SourceIP,*), TotalBytesSentinMB = sum(TotalSentBytesinMB), TotalBytesReceivedinMB = sum(TotalReceivedBytesinMB), SourceIPList = make_set(SourceIP), DestinationIPList = make_set(DestinationIPList) by AnomalyHour,TotalBytesSentinMBperHour, baselinebytessentperHour, score, anomalies  
+  | project DeviceVendor, AnomalyHour,TimeGeneratedMax, SourceIPMax, SourceIPList, DestinationIPList, DestinationPortList, TotalBytesSentinMB, TotalBytesReceivedinMB, TotalBytesSentinMBperHour, baselinebytessentperHour, score, anomalies 
```

Comment:
Any reason we need a project here and not just going with the summarize results above?

Reply:
same reason as i stated for performance reasons